src/api/user/user.router.ts

```diff
@@ -0,0 +1,90 @@
+import express, { NextFunction } from "express";
+import type { Request, Response } from "express";
+import { catchErrors, validateUuidParam } from "../../utils/middleware";
+import {
+  deleteUser,
+  getUser,
+  getUserBySystemId,
+  getUsers,
+  updateUser,
+  upsertUser,
+} from "./user.service";
+import { apiError } from "../../utils/types";
+
+export const userRouter = express.Router();
+
+/**
+ ** User Router Home
+ */
+userRouter.get(
+  "/",
+  catchErrors(async (req: Request, res: Response) => {
+    const users = await getUsers();
+    return res.status(200).json(users);
+  })
+);
+
+/**
+ ** Create new user
+ */
+userRouter.post(
+  "/create",
+  catchErrors(async (req: Request, res: Response) => {
+    const userData = req.body;
+    const newUser = await upsertUser(userData);
+    return res.status(201).json(newUser);
+  })
+);
+
+/**
+ ** All user_id related routes
+ */
+userRouter
+  .route("/:id")
+  .all(
+    catchErrors(async (req: Request, res: Response, next: NextFunction) => {
+      // validate user id and confirm that user exists
+      const id = validateUuidParam(req);
+      const user = await getUser(id);
```

Comment:
Wanna try implementing something like this for getUser(id)
So we dont have to call getUser twice per request
https://stackoverflow.com/questions/18875292/passing-variables-to-the-next-middleware-using-next-in-express-js